### DISCUSSION TITLE
Controller?

### DISCUSSION THREAD

**Discussion by mjkatona** [RaspAP/raspap-webgui#786] [General]
Hey there. Stumbled on this, was reading but couldn't find answers in the docs or I just didn't look hard enough.. BUT...

Say I set up 4 RPIs as Bridge/APs in a house.. Are these designed to connect back to a controller or do each AP need to be touched? 

In the same design thought-- If all APs are running ESS or something like that, could they auto-adjust their RF-Power levels based on peer APs with the same SSID? For example: AP-1 is basement, AP-2 is first floor, and AP-3 is second floor all using RASPAP as the SSID. AP-1 and AP-2 "SSID bubble" overlaps and so they start to drop their RF-Levels until they have the least amount of over-lap. This would create "the perfect bubble" .. 

THanks for listening.

**Comment by billz** [comment]
Interesting question. There is no controller facility, so each AP would need to be touched.
However... assuming each AP broadcasts the same SSID, peer signal levels could be detected with `iw dev wlan0 scan` and used to adjust txpower. e.g.,

```
$ sudo iw dev wlan0 set txpower fixed 2500
$ iw dev wlan0 info
Interface wlan0
	ifindex 3
	wdev 0x1
	addr [mac]
	ssid RaspAP
	type AP
	wiphy 0
	channel 36 (5180 MHz), width: 80 MHz, center1: 5210 MHz
	txpower 25.00 dBm
```
(works on the Pi4)

**Reply by mjkatona** [reply]
I plan on downloading and messing around with it this weekend (I have 10 pi's LOL) .. Is there a flat file for the settings? Here is my thought, at least in my network .. I have a remote mount on a NAS which I point all my PIs at.. So I install RASPAP on a few and have them all point to the share for their config file based on hostname or mac .. I then can write a perl script on the APs to check their config file on the /share/rasap/ap/configs and if it's different then the one running local to the AP, pull the new one and restart the AP.

I think if this works, THEN build hooks into the DHCP server (I dont know how yet) looking for the MAC of the AP with the thought when it first boots up, the DHCP server can push the config. 

Just thinking out loud in the morning before work :)

**Reply by billz** [reply]
.json files in `/config` are used for default values, although RaspAP writes dnsmasq & dhcp settings to the config files directly. So there's no monolithic flat file. More on this [here](https://docs.raspap.com/defaults/) and [here](https://github.com/billz/raspap/releases/tag/2.6-beta).

A sync script like you describe would certainly work. Re: RF-Power levels, I was thinking along the lines of a process on each Pi that would poll nearby peers and auto adjust TX power dynamically. Possibly related:

https://ieeexplore.ieee.org/document/6952281
https://ieeexplore.ieee.org/document/1651683

Beyond this, I think the controller AP idea has merit.

**Reply by billz** [reply]
In progress... 

<img width="399" alt="txpower" src="https://user-images.githubusercontent.com/229399/115365187-6e3bb200-a1c4-11eb-889b-7a777481cfa1.png">

**Comment by dingausmwald** [comment]
Hey there. Got directed here. Hope my question doesnt bother.
Could finally set up my raspzero (rasp os) with wifi client ap mode after several retries. Even dhcp is properly used from main router.

My question:
Can i use the same ssid as the main router and my devices decide which network they take? Or is some mesh technique like batman adv necessary? I want to extend my wifi to reach some wifi sockets and i dont want to change the settings for every socket and do the testing which wifi for which socket is the best.
